### PR TITLE
Simd int64 mod

### PR DESCRIPTION
**Pull request by P1K**
Tested with AVX, AVX2, AVX512.
Cf. issue #224.

### CODE REVIEW

File: fflas-ffpack/fflas/fflas_simd/simd128_int64.inl [hunk 1]
```diff
@@ -687,38 +687,40 @@ INLINE CONST vect_t Simd128_impl<true,true,true,8>::mulhi_fast(vect_t x, vect_t
...
 
-// warning : may be off by 1 multiple, but we save a mul...
-template <bool overflow, bool poweroftwo, int8_t shifter>
-INLINE CONST vect_t Simd128_impl<true,true,true,8>::mod(vect_t &C, const vect_t &P, const vect_t &magic, const vect_t &NEGP,
-                                                        const vect_t &MIN, const vect_t &MAX, vect_t &Q, vect_t &T) {
-#ifdef __INTEL_COMPILER
-    // Works fine with ICC 15.0.1 - A.B.
-    // #warning "not tested"
-    C = _mm_rem_epi64(C, P);
+// FIXME why cannot use Simd128<double>::vect_t in the declaration instead of __m128d?
```
Comment (ClementPernet) [inline fflas-ffpack/fflas/fflas_simd/simd128_int64.inl]: I have the same question? is it just not possible or did you just prefer the simpler call to __m128d ?

Comment (P1K) [reply fflas-ffpack/fflas/fflas_simd/simd128_int64.inl]: I just couldn't find the right syntax to make it work, but I'm probably just bad at C++.

Comment (ClementPernet) [reply fflas-ffpack/fflas/fflas_simd/simd128_int64.inl]: Ok, neither can I after a 5min attempt. Fine to let it as is.

**Comment by ClementPernet** [comment]
LGTM

**Pull request merged**